sdk/README.md

```diff
@@ -0,0 +1,334 @@
+# SDK
+
+Smithy's SDK.
+
+## Component
+
+The component package can be used to write Smithy
+components that represents
+Vulnerability Findings data in [OCSF](https://docs.aws.amazon.com/security-lake/latest/userguide/open-cybersecurity-schema-framework.html) format.
+
+OCSF is a standard for representing vulnerability reports that can be understood by a variety of
+security tools.
+
+This package allows you to focus on writing the business logic for your component
+while taking care of the boring things for you:
+
+* running the components steps in a predictable and reliable way
+* handle intricacies like cancellations and graceful shutdown
+* taking care of logging and panic handling
+* reporting common metrics to track what your component is doing
+
+You can customise a component using the following environment variables:
+
+| Environment Variable       | Type   | Required | Possible Values                |
+|----------------------------|--------|----------|--------------------------------|
+| SMITHY\_COMPONENT\_NAME       | string | yes      | -                              |
+| SMITHY\_LOGGING\_LOG\_LEVEL    | string | false    | info, debug, warn, error       |
+
+Or you can use the supplied options `RunOption`, like in this example:
+
+```go
+component.RunTarget(
+	ctx, 
+	sampleTarget{}, 
+	component.RunnerWithComponentName("sample-target"),
+)
+```
+
+### Components
+
+#### Target
+
+A `Target` component should be used to prepare a target for scanning.
+
+For example, cloning a repository and make it available for a `Scanner` to scan.
```

Comment:
It would be clearer to give an example of an existing component. e.g. "Git-clone" is a target component because it clones a repository and prepares it for scanning.